snuba/subscriptions/data.py

```diff
@@ -94,49 +94,68 @@ def add_conditions(
+        else:
             raise InvalidSubscriptionError(
-                "Entity must have a timestamp column for subscriptions"
+                "Only simple queries and join queries are supported"
             )
-
-        conditions_to_add: Sequence[Expression] = [
-            binary_condition(
-                ConditionFunctions.EQ,
-                Column(None, None, "project_id"),
-                Literal(None, self.project_id),
-            ),
-            binary_condition(
-                ConditionFunctions.GTE,
-                Column(None, None, required_timestamp_column),
-                Literal(None, (timestamp - timedelta(seconds=self.time_window_sec))),
-            ),
-            binary_condition(
-                ConditionFunctions.LT,
-                Column(None, None, required_timestamp_column),
-                Literal(None, timestamp),
-            ),
-        ]
-
-        new_condition = combine_and_conditions(conditions_to_add)
-        condition = query.get_condition()
-        if condition:
-            new_condition = binary_condition(
-                BooleanFunctions.AND, condition, new_condition
+        for alias, entity in entities:
+            conditions_to_add: Sequence[Expression] = [
+                binary_condition(
+                    ConditionFunctions.EQ,
+                    Column(None, alias, "project_id"),
+                    Literal(None, self.project_id),
+                ),
+            ]
+
+            required_timestamp_column = entity.required_time_column
+            if required_timestamp_column is not None:
+                added_timestamp_column = True
+                conditions_to_add.extend(
+                    [
+                        binary_condition(
+                            ConditionFunctions.GTE,
+                            Column(None, alias, required_timestamp_column),
+                            Literal(
+                                None,
+                                (timestamp - timedelta(seconds=self.time_window_sec)),
+                            ),
+                        ),
+                        binary_condition(
+                            ConditionFunctions.LT,
+                            Column(None, alias, required_timestamp_column),
+                            Literal(None, timestamp),
+                        ),
+                    ]
+                )
+
+            new_condition = combine_and_conditions(conditions_to_add)
+            condition = query.get_condition()
+            if condition:
+                new_condition = binary_condition(
+                    BooleanFunctions.AND, condition, new_condition
+                )
+
+            query.set_ast_condition(new_condition)
+
+            subscription_processors = self.entity.get_subscription_processors()
+            if subscription_processors:
+                for processor in subscription_processors:
+                    processor.process(query, self.metadata, offset)
```

Comment:
I think you'll need to rework the processors here, because I suspect they expect simple queries instead of `CompositeQuery`. If any of the processors use the from clause you'll need to refactor them as well.

Reply:
The only processor here is https://github.com/getsentry/snuba/blob/7d2de78aac9e8c6948b6c0c57c28c3245fac6f0f/snuba/datasets/entity_subscriptions/processors.py#L40. It expects